### PR TITLE
XWIKI-18992: Replace the Livetable of likers of a page with a Live Data macro

### PR DESCRIPTION
**Pull request by manuelleduc**
Jira: https://jira.xwiki.org/browse/XWIKI-18992

**Before**

image

**After**

image

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-like/xwiki-platform-like-api/src/main/resources/templates/likers.vm [hunk 1]
```diff
@@ -17,106 +17,118 @@
...
+    "rows": $userRows
+  })
+  #livetable_filterObfuscated($map)
+  #jsonResponse($map)
+#end
+{{/velocity}}{{velocity}}##A line break is inserted if the two macro are not closing and opening on the same line.
+  #if ($request.livetable == "true")
+    #break
+  #end
+  = $escapetool.xml($services.localization.render('like.likers.title', [$doc.plainTitle])) =
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-like/xwiki-platform-like-api/src/main/resources/templates/likers.vm:108]: This seems like the wrong escaping for XWiki syntax.

File: xwiki-platform-core/xwiki-platform-like/xwiki-platform-like-api/src/main/resources/templates/likers.vm [hunk 2]
```diff
@@ -17,106 +17,118 @@
...
+  #macro (displayUserAliasWithAvatar $userReference $disabled)
     <div class="user#if ($disabled) disabled#end" data-reference="$escapetool.xml($userReference)">
     <span class="user-avatar-wrapper">
       #getUserAvatarURL($userReference $avatarURL 120)
       <img class="user-avatar" src="$escapetool.xml($avatarURL.url)" />
     </span>
       <a href="$xwiki.getURL($userReference)">$escapetool.xml($userReference.name)</a>
     </div>
+  #end
+  $response.setContentType('application/json')
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-like/xwiki-platform-like-api/src/main/resources/templates/likers.vm:32]: This shouldn't be necessary when using `#jsonResponse($map)`.

File: xwiki-platform-core/xwiki-platform-like/xwiki-platform-like-api/src/main/resources/templates/likers.vm [hunk 3]
```diff
@@ -17,106 +17,118 @@
...
+  #set ($likeRecords = $services.like.getLikers($documentReference, $offset, $limit))
+  #set ($userRows = [])
+  #foreach($userReference in $likeRecords)
+    #set ($grayed = $xcontext.userReference == $userReference.reference)
+    #set ($userDoc = $xwiki.getDocument($userReference.reference))
+    #set ($userProperties = $services.user.getProperties($userReference))
+    #set ($userObject = $user.getObject('XWiki.XWikiUsers'))
+    #if ($services.security.authorization.hasAccess('view', $userDoc.documentReference))
+      #set ($row = {
+        'grayed': $grayed,
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-like/xwiki-platform-like-api/src/main/resources/templates/likers.vm:66]: Is this supported by LiveData?

**Review by michitux — CHANGES_REQUESTED** [review]
Overall, this looks good, some small comments.

File: xwiki-platform-core/xwiki-platform-like/xwiki-platform-like-api/src/main/resources/templates/likers.vm (hunk 3 above)
Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-like/xwiki-platform-like-api/src/main/resources/templates/likers.vm:66]: It is not, but in the case of this Livetable, the property is never used. I'll check if this is caused by another regression.

File: xwiki-platform-core/xwiki-platform-like/xwiki-platform-like-api/src/main/resources/templates/likers.vm (hunk 1 above)
Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-like/xwiki-platform-like-api/src/main/resources/templates/likers.vm:108]: Thanks

**Pull request merged**